lib/fauna/connection.rb

```diff
@@ -1,90 +1,127 @@
+    attr_reader :credentials
+    # Read timeout in seconds.
+    attr_reader :timeout
+    # \Connection open timeout in seconds.
+    attr_reader :connection_timeout
+    # Faraday adapter in use.
+    attr_reader :adapter
+    # List of loggers to record request/response data to.
+    attr_reader :logger
+
+    ##
+    # Creates a new Connection object to be used in the creation of a FaunaDB client.
+    #
+    # +params+:: A list of parameters to configure the connection with.
+    #            +:logger+:: A logger to output client traffic to.
+    #                        Setting the +FAUNA_DEBUG+ environment variable will also log to +STDERR+.
+    #            +:domain+:: The domain to send requests to.
+    #            +:scheme+:: Scheme to use when sending requests (either +http+ or +https+).
+    #            +:port+:: Port to use when sending requests.
+    #            +:timeout+:: Read timeout in seconds.
+    #            +:connection_timeout+:: \Connection open timeout in seconds.
+    #            +:adapter+:: Faraday adapter to use. Either can be a symbol for the adapter, or an array of arguments.
+    #            +:secret+:: Credentials to use when sending requests. User and pass must be separated by a colon.
     def initialize(params = {}) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
-      @logger = params[:logger] || nil
+      @logger = []
       @domain = params[:domain] || 'rest.faunadb.com'
       @scheme = params[:scheme] || 'https'
       @port = params[:port] || (@scheme == 'https' ? 443 : 80)
       @timeout = params[:timeout] || 60
       @connection_timeout = params[:connection_timeout] || 60
       @adapter = params[:adapter] || Faraday.default_adapter
-      @credentials = params[:secret].to_s.split(':')
+      @credentials = params[:secret].to_s.split(':', 2)
+
+      @logger.push params[:logger] unless params[:logger].nil?
 
       if ENV['FAUNA_DEBUG']
-        @logger = Logger.new(STDERR)
-        @logger.formatter = proc { |_, _, _, msg| "#{msg}\n" }
+        debug_logger = Logger.new(STDERR)
+        debug_logger.formatter = proc { |_, _, _, msg| "#{msg}\n" }
+        @logger.push debug_logger
       end
 
-      @conn = Faraday.new(
-        :url => "#{@scheme}://#{@domain}:#{@port}/",
-        :headers => { 'Accept-Encoding' => 'gzip', 'Content-Type' => 'application/json;charset=utf-8' },
-        :request => { :timeout => @timeout, :open_timeout => @connection_timeout },
+      # Create connection
+      @connection = Faraday.new(
+          url: "#{@scheme}://#{@domain}:#{@port}/",
+          headers: { 'Accept-Encoding' => 'gzip,deflate', 'Content-Type' => 'application/json;charset=utf-8' },
+          request: { timeout: @timeout, open_timeout: @connection_timeout },
       ) do |conn|
-        conn.adapter(@adapter)
+        # Let us specify arguments so we can set stubs for test adapter
+        if @adapter.is_a? Symbol
```

Comment:
this seems fragile. are you trying to splat out adapter if its an array? if so, use `*Array(adapter)`